src/server/game/Entities/Creature/TemporarySummon.h

```diff
@@ -46,7 +46,8 @@ class TC_GAME_API TempSummon : public Creature
         void RemoveFromWorld() override;
         void SetTempSummonType(TempSummonType type);
         void SaveToDB(uint32 /*mapid*/, uint8 /*spawnMask*/, uint32 /*phaseMask*/) override { }
-        Unit* GetSummoner() const;
+        WorldObject* GetSummoner() const;
+        Unit* GetSummonerUnit() const;
```

Comment:
Might as well add `GetSummonerGameObject()` for completeness sake and to more clearly communicate that summoner might be one of these two types.